src/nft/components/layout/Tabs.tsx

```diff
@@ -0,0 +1,101 @@
+import React, {
+  cloneElement,
+  createContext,
+  HTMLAttributes,
+  isValidElement,
+  ReactNode,
+  useContext,
+  useMemo,
+  useState,
+} from 'react'
+
+function useConstant<T>(fn: () => T): T {
+  const ref = React.useRef<{ v: T }>()
+
+  if (!ref.current) ref.current = { v: fn() }
+
+  return ref.current.v
+}
+
+const TabsState = createContext<[number, (n: number) => void]>([0, () => void 0])
+const Elements = createContext<{ tabs: React.ReactNode[]; panels: React.ReactNode[] }>({
+  tabs: [],
+  panels: [],
+})
+
+export const Tabs: React.FC<{ children: ReactNode; state?: [number, (n: number) => void] }> = ({
```

Comment:
i don't think you need to define the type for the react fc? it's redundant with arguments inside already